src/main/kotlin/org/neo4j/graphql/SchemaBuilder.kt

```diff
@@ -99,8 +100,8 @@ object SchemaBuilder {
         return handler
     }
 
-    private fun augmentSchema(sourceSchema: GraphQLSchema, handler: List<AugmentationHandler>): GraphQLSchema {
-        val types = sourceSchema.typeMap.toMutableMap()
+    private fun augmentSchema(sourceSchema: GraphQLSchema, handler: List<AugmentationHandler>, config: SchemaConfig): GraphQLSchema {
+        val types = sourceSchema.typeMap.toMap(ConcurrentHashMap())
```

Comment:
why do we need the CHM?

Reply:
Wen we iterate over the types, the ordering-type may be added so it will throw an concurrent modification exception. This is a simple solution. Otherwise I need to split the logic up to first generate all required types and then enhance the types in a second run.

Reply:
yeah that would be better, can you create an issue so that we don't forget to fix this in the future?